src/plugins/unified_search/public/dataview_picker/change_dataview.test.tsx

```diff
@@ -165,14 +165,16 @@ describe('DataView component', () => {
     expect(props.onTextLangQuerySubmit).toHaveBeenCalled();
   });
 
-  it('should not propagate the adHoc dataviews for text based mode', async () => {
+  const runAdHocDataViewTest = (esqlMode: boolean) => {
     const component = mount(
       wrapDataViewComponentInContext(
         {
           ...props,
           onDataViewCreated: jest.fn(),
-          textBasedLanguages: [TextBasedLanguages.ESQL, TextBasedLanguages.SQL],
-          textBasedLanguage: TextBasedLanguages.ESQL,
+          ...(esqlMode && {
+            textBasedLanguages: [TextBasedLanguages.ESQL, TextBasedLanguages.ESQL],
```

Comment:
```suggestion
            textBasedLanguages: [TextBasedLanguages.ESQL],
```

Reply:
I blame find and replace 🙂 Fixed in this commit: https://github.com/elastic/kibana/pull/177109/commits/fccf8a70dc7707dfa388241008269c7c4840ffdd.